File: 144-binary-tree-preorder-traversal/binary-tree-preorder-traversal.swift
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public var val: Int
 *     public var left: TreeNode?
 *     public var right: TreeNode?
 *     public init() { self.val = 0; self.left = nil; self.right = nil; }
 *     public init(_ val: Int) { self.val = val; self.left = nil; self.right = nil; }
 *     public init(_ val: Int, _ left: TreeNode?, _ right: TreeNode?) {
 *         self.val = val
 *         self.left = left
 *         self.right = right
 *     }
 * }
 */
class Solution {
    func preorderTraversal(_ root: TreeNode?) -> [Int] {
        func preorder(root curr: TreeNode?, array arr: inout [Int]) -> [Int]{
            guard let curr = curr else {return []}

            
            arr.append(curr.val)
            preorder(root : curr.left, array: &arr)
            preorder(root: curr.right,array: &arr)
            return arr
        }
        var ans: [Int] = [Int]()

        return preorder(root: root, array: &ans)
    }
    
}

File: 144-binary-tree-preorder-traversal/binary-tree-preorder-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def preorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        ans = []
        def re(node, ans):
            if not node:
                return
            ans.append(node.val)
            re(node.left, ans)
            re(node.right, ans)
            return ans
        return re(root, ans)